## Final Project Submission

Please fill out:
* Student name: 
* Student pace: self paced / part time / full time
* Scheduled project review date/time: 
* Instructor name: 
* Blog post URL:


Relationship between reviews and revenue for top 100 films. 


Following data is from Rotten Tomatoes Reviews and The Numbers grossing revenue. 




In [1]:
# Your code here - remember to use markdown cells for comments as well!

In [2]:
import pandas as pd

In [3]:
import gzip

Import data for reviews from Rotten Tomatoes

In [4]:
rt = pd.read_csv(gzip.open('zippedData/rt.reviews.tsv.gz', 'rt',encoding='latin-1'), sep='\t')

In [5]:
rt.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54432 entries, 0 to 54431
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   id          54432 non-null  int64 
 1   review      48869 non-null  object
 2   rating      40915 non-null  object
 3   fresh       54432 non-null  object
 4   critic      51710 non-null  object
 5   top_critic  54432 non-null  int64 
 6   publisher   54123 non-null  object
 7   date        54432 non-null  object
dtypes: int64(2), object(6)
memory usage: 3.3+ MB


In [6]:
rt.head()

id                                             review rating   fresh  \
0   3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1   3  It's an allegory in search of a meaning that n...    NaN  rotten   
2   3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3   3  Continuing along a line introduced in last yea...    NaN   fresh   
4   3             ... a perverse twist on neorealism...     NaN   fresh   

           critic  top_critic         publisher               date  
0      PJ Nabarro           0   Patrick Nabarro  November 10, 2018  
1  Annalee Newitz           0           io9.com       May 23, 2018  
2    Sean Axmaker           0  Stream on Demand    January 4, 2018  
3   Daniel Kasman           0              MUBI  November 16, 2017  
4             NaN           0      Cinema Scope   October 12, 2017

In [7]:
rt.tail()

id                                             review rating   fresh  \
54427  2000  The real charm of this trifle is the deadpan c...    NaN   fresh   
54428  2000                                                NaN    1/5  rotten   
54429  2000                                                NaN    2/5  rotten   
54430  2000                                                NaN  2.5/5  rotten   
54431  2000                                                NaN    3/5   fresh   

                  critic  top_critic        publisher                date  
54427      Laura Sinagra           1    Village Voice  September 24, 2002  
54428  Michael Szymanski           0       Zap2it.com  September 21, 2005  
54429       Emanuel Levy           0  EmanuelLevy.Com       July 17, 2005  
54430   Christopher Null           0   Filmcritic.com   September 7, 2003  
54431    Nicolas Lacroix           0     Showbizz.net   November 12, 2002

In [8]:
pd.set_option("display.max_rows", None)
rt['rating'].value_counts()

3/5       4327
4/5       3672
3/4       3577
2/5       3160
2/4       2712
2.5/4     2381
3.5/4     1777
3.5/5     1289
5/5       1237
B         1163
1/5       1113
1.5/4     1095
4/4        995
2.5/5      992
B+         832
1/4        822
B-         821
C          779
C+         665
4.5/5      567
7/10       522
A-         514
8/10       505
C-         493
6/10       468
1.5/5      405
A          397
5/10       351
D          324
9/10       304
4/10       252
D+         204
0/5        162
8          143
3/10       140
1          138
0/4        132
7          125
F          109
0.5/4      100
D-          84
6           82
A+          73
2/10        73
7.5/10      58
0.5/5       52
5           48
8.5/10      41
6.5/10      39
4           34
3           33
9           33
1/10        30
5.5/10      27
3.0/4       21
4.5/10      21
3.5         18
9.5/10      17
3/6         14
3.0/5       12
2.0/5       12
4.0/4       11
2           11
5.0/5        9
4/6          9
2.1/2        9
6/8       

In [9]:
pd.reset_option("display.max_rows")

Noticed that many of the reviews are either strings such as B+ or C, or a division such as 2/5 or 1/10 or 1/4. 

Will replace values within each of their own categories to give them a new review grade of:

A 100-90% , B 80-89% , C 70-79% , D 60-69%, F 0-59%

A is any high rating at or above a 90% score
B is any middle rating such as 80-89%
C is any low-middle rating such as 70-79%
D is any low rating from 60-69%
F is rating below a 59% and lower


Will sort by top_critic reviews and civilian_reviews

In [10]:
#Deleting the characters of '-' and '+' from column rating

rt['rating'] = rt['rating'].str.replace('-', '').str.replace('+', '')

In [11]:
rt['rating'].isna().sum()

13517

Changing the value of the missing values because there are 13,517 missing within 54,431 rows within the DataFrame.


In [25]:
#Following code changes non-numeric object values to floats then will convert all values to non-numeric values


    if grade == 'A':
        return 95.0
    elif grade == 'B':
        return 85.0
    elif grade == 'C':
        return 75.0
    elif grade == 'D':
        return 65.0
    elif grade == 'F':
        return 0
    else:
        return None


IndentationError: unexpected indent (<ipython-input-25-8cdc7eea6504>, line 4)

In [20]:

    if rating >= 90:
        return 'A'
    elif rating >= 80:
        return 'B'
    elif rating >= 70:
        return 'C'
    elif rating >= 60:
        return 'D'
    else:
        return 'F'


TypeError: '>=' not supported between instances of 'str' and 'int'

The following code will now replace the median of rating values to each value of NaN.

In [36]:
mean = rt['rating'].mean()
rt['rating'].fillna(mean, inplace=True)

In [37]:
rt['rating'].head()

0    60.0
1     6.0
2     6.0
3     6.0
4     6.0
Name: rating, dtype: float64

In [ ]:
# Following code is a loop to push the grading scale stated above to the rating column 



Will now work on the rating column. 

Renaming the column of rating to keep the rating margin. 

Changing rating values to integers to lead to final analysis

In [30]:
rt = rt.rename(columns={'rating': 'rating (%)'})
rt

id                                             review  \
0         3  A distinctly gallows take on contemporary fina...   
1         3  It's an allegory in search of a meaning that n...   
2         3  ... life lived in a bubble in financial dealin...   
3         3  Continuing along a line introduced in last yea...   
4         3             ... a perverse twist on neorealism...    
...     ...                                                ...   
54427  2000  The real charm of this trifle is the deadpan c...   
54428  2000                                                NaN   
54429  2000                                                NaN   
54430  2000                                                NaN   
54431  2000                                                NaN   

      rating out of 5   fresh             critic  top_critic  \
0                 3/5   fresh         PJ Nabarro           0   
1                 NaN  rotten     Annalee Newitz           0   
2                 NaN   fresh       Sean Axmaker           0   
3                 NaN   fresh      Daniel Kasman           0   
4                 NaN   fresh                NaN           0   
...               ...     ...                ...         ...   
54427             NaN   fresh      Laura Sinagra           1   
54428             1/5  rotten  Michael Szymanski           0   
54429             2/5  rotten       Emanuel Levy           0   
54430           2.5/5  rotten   Christopher Null           0   
54431             3/5   fresh    Nicolas Lacroix           0   

              publisher                date  
0       Patrick Nabarro   November 10, 2018  
1               io9.com        May 23, 2018  
2      Stream on Demand     January 4, 2018  
3                  MUBI   November 16, 2017  
4          Cinema Scope    October 12, 2017  
...                 ...                 ...  
54427     Village Voice  September 24, 2002  
54428        Zap2it.com  September 21, 2005  
54429   EmanuelLevy.Com       July 17, 2005  
54430    Filmcritic.com   September 7, 2003  
54431      Showbizz.net   November 12, 2002  

[54432 rows x 8 columns]

Now to change rating column into integer.

In [11]:
#change string type to integer
rt["rating out of 5"] = rt["rating out of 5"].astype(int)

KeyError: 'rating out of 5'

Import data for revenue from The Numbers

In [18]:
df6 = pd.read_csv('zippedData/tn.movie_budgets.csv.gz')

In [19]:
df6.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747